### DISCUSSION TITLE
How to publish nextjs project

### DISCUSSION THREAD

**Discussion by Rishu2807** [community/community#64129] [New to GitHub]
I am getting error during build process , I modified my nextjs.yml file also . I am listing out my modified code in .yml file and error occurred during build process

#first modified code line 
echo "manager=yarn" >> ${{ github.workspace }}/.github_output
echo "command=install" >> ${{ github.workspace }}/.github_output
echo "runner=yarn" >> ${{ github.workspace }}/.github_output

# First error during 

" 0s
Run if [ -f "/home/runner/work/rishu.portfolio28/rishu.portfolio28/yarn.lock" ]; then
Unable to determine package manager
Error: Process completed with exit code 1. " 

# second modification 
- name: Detect package manager
  id: detect-package-manager
  run: |
    if [ -f "${{ github.workspace }}/yarn.lock" ]; then
      echo "manager=yarn" >> $GITHUB_OUTPUT
      echo "command=install" >> $GITHUB_OUTPUT
      echo "runner=yarn" >> $GITHUB_OUTPUT
      exit 0
    elif [ -f "${{ github.workspace }}/package.json" ]; then
      echo "manager=npm" >> $GITHUB_OUTPUT
      echo "command=ci" >> $GITHUB_OUTPUT
      echo "runner=npx --no-install" >> $GITHUB_OUTPUT
      exit 0
    else
      echo "Unable to determine package manager"
      exit 1
    fi
    
   # error during the build process
   
   Run if [ -f "/home/runner/work/rishu.portfolio28/rishu.portfolio28/yarn.lock" ]; then
Unable to determine package manager
Error: Process completed with exit code 1.

# third modification 
- name: Install Dependencies
  run: |
    if [ -f "yarn.lock" ]; then
      yarn install
    elif [ -f "package.json" ]; then
      npm ci
    else
      echo "Unable to determine package manager"
      exit 1
    fi
    
    
    - name: Build with Next.js
  run: npx next build
    
    
    # error 
    
    Run if [ -f "yarn.lock" ]; then
  if [ -f "yarn.lock" ]; then
    yarn install
  elif [ -f "package.json" ]; then
    npm ci
  else
    echo "Unable to determine package manager"
    exit 1
  fi
  shell: /usr/bin/bash -e {0}
Unable to determine package manager
Error: Process completed with exit code 1.

**Comment by vacisdev** [comment]
It seems the script is unable to detect the package manager, make sure the paths to `yarn.lock` or `package.json` are correct, and the respective files exist in the expected directory. If the issue persists, please provide further details on the directory structure and context of the build process.

**Reply by Rishu2807** [reply]
This is my by default configured nextjs.yml file . And I am using package.json . So what should be actual modification I have to do ? 
Can you please suggest , it will be really appreciable ? 



# Sample workflow for building and deploying a Next.js site to GitHub Pages
#
# To get started with Next.js see: https://nextjs.org/docs/getting-started
#
name: Deploy Next.js site to Pages

on:
  # Runs on pushes targeting the default branch
  push:
    branches: ["feature_rishu_portfolio"]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
permissions:
  contents: read
  pages: write
  id-token: write

# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
concurrency:
  group: "pages"
  cancel-in-progress: false

jobs:
  # Build job
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3
      - name: Detect package manager
        id: detect-package-manager
        run: |
          if [ -f "${{ github.workspace }}/yarn.lock" ]; then
            echo "manager=yarn" >> $GITHUB_OUTPUT
            echo "command=install" >> $GITHUB_OUTPUT
            echo "runner=yarn" >> $GITHUB_OUTPUT
            exit 0
          elif [ -f "${{ github.workspace }}/package.json" ]; then
            echo "manager=npm" >> $GITHUB_OUTPUT
            echo "command=ci" >> $GITHUB_OUTPUT
            echo "runner=npx --no-install" >> $GITHUB_OUTPUT
            exit 0
          else
            echo "Unable to determine package manager"
            exit 1
          fi
      - name: Setup Node
        uses: actions/setup-node@v3
        with:
          node-version: "16"
          cache: ${{ steps.detect-package-manager.outputs.manager }}
      - name: Setup Pages
        uses: actions/configure-pages@v3
        with:
          # Automatically inject basePath in your Next.js configuration file and disable
          # server side image optimization (https://nextjs.org/docs/api-reference/next/image#unoptimized).
          #
          # You may remove this line if you want to manage the configuration yourself.
          static_site_generator: next
      - name: Restore cache
        uses: actions/cache@v3
        with:
          path: |
            .next/cache
          # Generate a new cache whenever packages or source files change.
          key: ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json', '**/yarn.lock') }}-${{ hashFiles('**.[jt]s', '**.[jt]sx') }}
          # If source files changed but packages didn't, rebuild from a prior cache.
          restore-keys: |
            ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json', '**/yarn.lock') }}-
      - name: Install dependencies
        run: ${{ steps.detect-package-manager.outputs.manager }} ${{ steps.detect-package-manager.outputs.command }}
      - name: Build with Next.js
        run: ${{ steps.detect-package-manager.outputs.runner }} next build
      - name: Static HTML export with Next.js
        run: ${{ steps.detect-package-manager.outputs.runner }} next export
      - name: Upload artifact
        uses: actions/upload-pages-artifact@v2
        with:
          path: ./out

  # Deployment job
  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    needs: build
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v2

**Comment by Rishu2807** [comment]
This is my by default configured nextjs.yml file . And I am using package.json . So what should be actual modification I have to do ?
Can you please suggest , it will be really appreciable ?

Sample workflow for building and deploying a Next.js site to GitHub Pages
To get started with Next.js see: https://nextjs.org/docs/getting-started
name: Deploy Next.js site to Pages

on:

Runs on pushes targeting the default branch
push:
branches: ["feature_rishu_portfolio"]

Allows you to run this workflow manually from the Actions tab
workflow_dispatch:

Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
permissions:
contents: read
pages: write
id-token: write

Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
concurrency:
group: "pages"
cancel-in-progress: false

jobs:

Build job
build:
runs-on: ubuntu-latest
steps:
- name: Checkout
uses: actions/checkout@v3
- name: Detect package manager
id: detect-package-manager
run: |
if [ -f "${{ github.workspace }}/yarn.lock" ]; then
echo "manager=yarn" >> $GITHUB_OUTPUT
echo "command=install" >> $GITHUB_OUTPUT
echo "runner=yarn" >> $GITHUB_OUTPUT
exit 0
elif [ -f "${{ github.workspace }}/package.json" ]; then
echo "manager=npm" >> $GITHUB_OUTPUT
echo "command=ci" >> $GITHUB_OUTPUT
echo "runner=npx --no-install" >> $GITHUB_OUTPUT
exit 0
else
echo "Unable to determine package manager"
exit 1
fi
- name: Setup Node
uses: actions/setup-node@v3
with:
node-version: "16"
cache: ${{ steps.detect-package-manager.outputs.manager }}
- name: Setup Pages
uses: actions/configure-pages@v3
with:
# Automatically inject basePath in your Next.js configuration file and disable
# server side image optimization (https://nextjs.org/docs/api-reference/next/image#unoptimized).
#
# You may remove this line if you want to manage the configuration yourself.
static_site_generator: next
- name: Restore cache
uses: actions/cache@v3
with:
path: |
.next/cache
# Generate a new cache whenever packages or source files change.
key: ${{ runner.os }}-nextjs-${{ hashFiles('/package-lock.json', '/yarn.lock') }}-${{ hashFiles('.[jt]s', '.[jt]sx') }}
# If source files changed but packages didn't, rebuild from a prior cache.
restore-keys: |
${{ runner.os }}-nextjs-${{ hashFiles('/package-lock.json', '/yarn.lock') }}-
- name: Install dependencies
run: ${{ steps.detect-package-manager.outputs.manager }} ${{ steps.detect-package-manager.outputs.command }}
- name: Build with Next.js
run: ${{ steps.detect-package-manager.outputs.runner }} next build
- name: Static HTML export with Next.js
run: ${{ steps.detect-package-manager.outputs.runner }} next export
- name: Upload artifact
uses: actions/upload-pages-artifact@v2
with:
path: ./out

Deployment job
deploy:
environment:
name: github-pages
url: ${{ steps.deployment.outputs.page_url }}
runs-on: ubuntu-latest
needs: build
steps:
- name: Deploy to GitHub Pages
id: deployment
uses: actions/deploy-pages@v2

**Comment by vacisdev** [comment]
Since you mentioned that you are using `package.json` and not `yarn.lock`, you can modify the detection part of the script to default to npm if `yarn.lock` is not found. The given script should already work with this setup, so make sure that the path to `package.json` is correct within your workspace, and that the `package.json` file is indeed present in the repository at the expected location. If everything seems correct, try running the build process again.

```
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - name: Checkout
      uses: actions/checkout@v3

    - name: Setup Node
      uses: actions/setup-node@v3
      with:
        node-version: "16"
        cache: npm

    - name: Install dependencies
      run: |
        if [ -f "package.json" ]; then
          npm ci
        else
          echo "Unable to determine package manager"
          exit 1
        fi

    - name: Build with Next.js
      run: npx next build

    - name: Static HTML export with Next.js
      run: npx next export

    - name: Upload artifact
      uses: actions/upload-pages-artifact@v2
      with:
        path: ./out
```

**Comment by gamal2001** [comment]
checkout this repo:
https://github.com/DaveAldon/Next.js-and-GitHub-Pages-Example
